src/main/java/org/gridsuite/modification/server/modifications/TwoWindingsTransformerModification.java

```diff
@@ -346,15 +333,48 @@ private void addRatioTapChangersToTwoWindingsTransformer(Network network, TwoWin
             ratioTapChangerAdder.setRegulating(ratioTapChangerInfos.getRegulating().getValue());
         } else if (ratioTapChanger != null) {
             ratioTapChangerAdder.setRegulating(ratioTapChanger.isRegulating());
         }
-        modifyRatioVoltageRegulation(ratioTapChangerInfos, twt, ratioTapChangerAdder, ratioTapChangerReporter, network);
-        addTapChangerPositionsAndSteps(ratioTapChangerInfos, ratioTapChanger, ratioTapChangerAdder, ratioTapChangerReporter);
-        ratioTapChangerAdder.add();
+        List<Report> voltageRegulationReports = new ArrayList<>();
+        modifyRatioVoltageRegulation(ratioTapChangerInfos, twt, ratioTapChangerAdder, voltageRegulationReports, network);
+        List<Report> positionsAndStepsReports = new ArrayList<>();
+        addTapChangerPositionsAndSteps(ratioTapChangerInfos, ratioTapChanger, ratioTapChangerAdder, positionsAndStepsReports);
+
+        // TODO : This is a workaround that needs to be removed when modifying tap changer steps will be possible
+        // workaround : we need to remove the tap changer before adding the new one to avoid having two tap changers on the same transformer
+        // and we need to prepare a backup in case the addition fails
+        RatioTapChangerAdder ratioTapChangerBackUp = null;
+        if (ratioTapChanger != null) {
+            ratioTapChangerBackUp = rollbackRatioModificationsAdder(twt, ratioTapChanger);
+            ratioTapChanger.remove();
+        }
+
+        try {
+            ratioTapChangerAdder.add();
+        } catch (Exception e) {
+            if (ratioTapChangerBackUp != null) {
+                ratioTapChangerBackUp.add();
+            }
+            throw e;
+        }
+
+        if (!ratioTapChangerReports.isEmpty() || !voltageRegulationReports.isEmpty() || !positionsAndStepsReports.isEmpty()) {
+            Reporter ratioTapChangerReporter = ModificationUtils.getInstance().reportModifications(subReporter, ratioTapChangerReports, TapChangerType.RATIO.name(), RATIO_TAP_CHANGER_SUBREPORTER_DEFAULT_MESSAGE);
+            if (ratioTapChangerReporter == null) {
+                ratioTapChangerReporter = subReporter.createSubReporter(TapChangerType.RATIO.name(), RATIO_TAP_CHANGER_SUBREPORTER_DEFAULT_MESSAGE);
+                ratioTapChangerReporter.report(Report.builder()
+                            .withKey(TapChangerType.RATIO.name())
+                            .withDefaultMessage(RATIO_TAP_CHANGER_SUBREPORTER_DEFAULT_MESSAGE)
+                            .withSeverity(TypedValue.INFO_SEVERITY)
+                            .build());
+            }
+            ModificationUtils.getInstance().reportModifications(ratioTapChangerReporter, voltageRegulationReports, "ratioTapChangerVoltageRegulationModification", "    Voltage regulation");
+            ModificationUtils.getInstance().reportModifications(ratioTapChangerReporter, positionsAndStepsReports, "ratioTapChangerPositionsAndStepsModification", "    Tap Changer");
+        }
```

Comment:
could you specify the scope of the code to be removed ?
do we we have to remove everything from TODO to the end of this function ?